src/pip/_internal/req/req_file.py

```diff
@@ -563,7 +565,56 @@ def get_file_content(url: str, session: "PipSession") -> Tuple[str, str]:
     # Assume this is a bare path.
     try:
         with open(url, "rb") as f:
-            content = auto_decode(f.read())
+            raw_content = f.read()
     except OSError as exc:
         raise InstallationError(f"Could not open requirements file: {exc}")
+
+    content = _decode_req_file(raw_content, url)
```

Comment:
Hmm, if the decoding with the locale encoding also fails, the exception won't be caught. Any objections to calling `_decode_req_file` within in the try-except?

Reply:
> Hmm, if the decoding with the locale encoding also fails, the exception won't be caught. Any objections to calling `_decode_req_file` within in the try-except?

No objection to adding it there. My original assumption, and the reason I moved it out, was the catching of `OSError` was only really relevant for reading the content, and not the actual decoding.